src/components/YouTube/useVideos.js

```diff
@@ -0,0 +1,40 @@
+import { useEffect, useState } from 'react';
+
+import mockData from './List/youtube-videos-mock.json';
+
+function useVideos({ search }) {
+  const [videos, setVideos] = useState([]);
+  const [isLoading, setLoading] = useState(false);
+  const [error, setError] = useState(null);
```

Comment:
Great, I like that you are handling the data, the loading state and the possible errors